sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala

```diff
@@ -246,6 +246,9 @@ private[hive] trait HiveInspectors {
    * Wraps with Hive types based on object inspector.
    */
   protected def wrapperFor(oi: ObjectInspector, dataType: DataType): Any => Any = oi match {
+    case _ if dataType.isInstanceOf[UserDefinedType[_]] =>
```

Comment:
Oh, actually, I forgot that this path is doing a pattern matching via `ObjectInspector` not with `DataType`. I just remember I put this first for this reason (we should check `dataType` regardless of `ObjectInspector` first).

I definitely can avoid this via fixing more than here but I guess it is fine as it is because this is not in the critical path.